apps/src/maze/maze.js

```diff
@@ -1170,9 +1168,14 @@ Maze.scheduleAnimations = function (singleStep) {
     }
 
     animateAction(actions[index], singleStep, timePerAction);
+
+    var command = actions[index] && actions[index].command;
+    var timeModifier = (skin.actionSpeedScale && skin.actionSpeedScale[command]) || 1;
+    var timeForThisAction = parseInt(timePerAction / timeModifier, 10);
```

Comment:
Cool, I was under the impression setTimeout required an integer but I guess it will accept a float.